t/op/list.t

```diff
@@ -275,3 +275,13 @@ EOS
     my $e = "1"; $e = "(1,$e)" for 1..100_000; $e = "() = $e"; eval $e;
     is $@, "", "SEGV in Perl_list";
 }
+
+# GH #23447 - ensure that future optimizations don't break behaviour
+{
+    sub f {} my $y = ("a" .. "d")[2, 3, f()];
+    is $y, d, 'Non-returning final sub element in list slice in scalar context';
```

Comment:
Same comments here as in review of `t/op/array.t`.  Place sub definition on its one line.  Explain why `d` is not quoted (here and in line 286 below).

Reply:
Thanks, fixed as previous.